examples/example_sumArray.cpp

```diff
@@ -0,0 +1,207 @@
+#include "yacx/main.hpp"
+
+#include <algorithm>
+#include <array>
+#include <cstdio>
+#include <ctime>
+#include <functional>
+#include <iostream>
+#include <iterator>
+#include <random>
+#include <string>
+
+using yacx::Headers, yacx::Header, yacx::Source, yacx::KernelArg, yacx::Kernel, yacx::Options, 
+yacx::Device,yacx::type_of, yacx::KernelTime, yacx::Devices, yacx::arg_type, yacx::load;
+
+//A function template to compare the values of two array elements.
+template<typename type>
+void compare(type *lhs, type *rhs, int array_size) {
+  //1. Declaring the variables prior to comparing 2 array elements
+  int errors{0};
+  float lhs_type, rhs_type;
+  std::string errors_string, array_size_string, i_string, lhs_string, 
+  rhs_string;
+
+  /*2A. Calculate the number of errors caused by inconsistencies between
+        values of two array elements.*/
+  for (int i{0}; i < array_size; i += 1) {
+    lhs_type=static_cast<int>(lhs[i]);
+    rhs_type=static_cast<int>(rhs[i]);
+
+    if ((lhs_type-rhs_type)!= 0) {
+      errors += 1;
+      i_string=std::to_string(i);
+      lhs_string=std::to_string(lhs[i]);
+      rhs_string=std::to_string(rhs[i]);
```

Comment:
I don't think this is necessary? I think << will cast an int to a string stream anyways?